src/modules/facility/dto/create-facility-request.dto.ts

```diff
@@ -0,0 +1,190 @@
+    example: '250',
+    maxLength: 3,
+  })
+  productTypeId: string;
+
+  @ValidatedStringApiProperty({
+    description: 'The name or description of the facility type.',
+    example: 'BOND',
+    maxLength: 13,
+  })
+  productTypeName: string;
+
+  @ValidatedStringApiProperty({
+    description: 'Credit Period',
+    example: '1',
+    maxLength: 12,
+  })
+  exposurePeriod: string;
+
+  @ValidatedCurrencyApiProperty({
+    description: 'Facility Currency Code e.g. CAD, USD.',
+  })
+  currency: string;
+
+  @ValidatedStringApiProperty({
+    description: 'The industry classification code of the obligor.',
+    example: '0116',
+    maxLength: 10,
+  })
+  obligorIndustryClassification: string;
+
+  @ValidatedDateOnlyApiProperty({
+    description: `The date from which the borrower can draw funds from the facility. If this is a future date then it will be replaced with today's date.`,
+  })
+  effectiveDate: DateOnlyString;
+
+  @ValidatedDateOnlyApiProperty({
+    description: 'The expiration date of the facility.',
+  })
+  guaranteeExpiryDate: DateOnlyString;
+
+  @ValidatedDateOnlyApiProperty({
+    description: 'The end date of the next quarter.',
+  })
+  nextQuarterEndDate: DateOnlyString;
+
+  @ValidatedNumberApiProperty({
+    description: 'The overall limit for the facility.',
+    minimum: 0,
+  })
+  maximumLiability: number;
+
+  @ValidatedPartyIdentifierApiProperty({
+    // TODO APIM-69: is this definitely a party id?
+    description: 'The party identifier of the Agent Bank for this facility.',
+  })
+  agentBankIdentifier: string;
+
+  @ValidatedStringApiProperty({
+    description: 'The code of the primary country of risk designated for this credit arrangement.',
+    example: EXAMPLES.COUNTRY_CODE,
+    maxLength: 3,
+  })
+  riskCountryCode: string;
+
+  @ValidatedStringApiProperty({
+    description: 'Pre-Issue Est. Payment Frequency QUARTERLY(2)',
+    example: '2',
+    maxLength: 1,
+  })
+  premiumFrequencyCode: string;
+
+  @ValidatedStringApiProperty({
+    description: 'CORPORATE(03)',
+    example: '03',
+    maxLength: 2,
+  })
+  riskStatusCode: string;
+
+  @ValidatedStringApiProperty({
+    description: 'Credit Review Risk Code', // TODO APIM-69: I think this is wrong, this field is actually used for OfficerRiskRatingTypeCode
+    example: '13',
+    maxLength: 2,
+  })
+  creditRatingCode: string;
+
+  @ValidatedStringApiProperty({
+    description: 'Case Stage this can be 06 Commitment and 07 Issued',
+    example: '07',
+    maxLength: 2,
+  })
+  facilityStageCode: string;
+
+  @ValidatedStringApiProperty({
+    description: 'Derive values A, M or N',
+    example: 'A',
+    maxLength: 4,
+  })
+  delegationType: string;
+
+  @ValidatedNumberApiProperty({
+    description: 'Bank Rate, this can be for Bond facility corresponding fee rate or for Loan/EWCS interest rate.',
+    minimum: 0,
+  })
+  interestOrFeeRate: number; // TODO APIM-69: document corrected typo
```

Comment:
I need to update our list of changes to say that I have correct the spelling of this field (it was `intrestOrFeeRate` before)